pkg/sql/crdb_internal.go

```diff
@@ -4379,3 +4380,61 @@ CREATE TABLE crdb_internal.cross_db_references (
 			})
 	},
 }
+
+var crdbLostTableDescriptors = virtualSchemaTable{
+	comment: `virtual table with table descriptors that still have data`,
+	schema: `
+CREATE TABLE crdb_internal.lost_descriptors_with_data (
+	descID
+		INTEGER NOT NULL
+);`,
+	populate: func(ctx context.Context, p *planner, dbContext *dbdesc.Immutable, addRow func(...tree.Datum) error) error {
+		maxDescIDKeyVal, err := p.extendedEvalCtx.DB.Get(context.Background(), p.extendedEvalCtx.Codec.DescIDSequenceKey())
+		if err != nil {
+			return err
+		}
+		maxDescID, err := maxDescIDKeyVal.Value.GetInt()
+		if err != nil {
+			return err
+		}
+		// Get all descriptors which will be used to determine
+		// which ones are missing
+		descriptors, err := catalogkv.GetAllDescriptorsUnvalidated(ctx, p.txn, p.extendedEvalCtx.Codec)
+		if err != nil {
+			return err
+		}
+		descriptorMap := make(map[descpb.ID]*catalog.Descriptor)
+		for _, desc := range descriptors {
+			descriptorMap[desc.GetID()] = &desc
+		}
+		// Loop over every possible descriptor ID
+		for id := int64(1); id < maxDescID; id++ {
+			// Skip over descriptors that are known
+			if _, ok := descriptorMap[descpb.ID(id)]; ok {
+				continue
+			}
+			// Generate a scan over the key space of the table to see
+			// if any data exists.
+			b := kv.Batch{}
+			prefix := p.extendedEvalCtx.Codec.TablePrefix(uint32(id))
```

Comment:
This didn't occur to me earlier but why not make `id` of type `uint32`? We don't really support IDs beyond 2^32 anyway.